src/services/modifyUser.service.ts

```diff
@@ -0,0 +1,50 @@
+import UserModel from '../models/user.model';
+import bcrypt from 'bcryptjs';
+
+const findUserByUsername = async (username: string) => {
+  return await UserModel.findOne({ where: { username } });
+};
+
+const deleteUserById = async (id: string) => {
+  return await UserModel.destroy({ where: { id } });
+};
+
+const updateUserById = async (
+  id: string,
+  {
+    username,
+    password,
+    address,
+    profile_photo,
+    mobNumber,
+    gender,
+    dob,
+    firstName,
+    lastName,
+  }: any
+) => {
+  const user = await UserModel.findByPk(id);
+  if (!user) {
+    throw new Error('User not found');
+  }
+
+  // Update the user's fields
+  user.username = username || user.username;
```

Comment:
username || user.username? please check the similar conditions too

Reply:
Resolved